BulletTrainClient.cs

```diff
@@ -10,14 +10,27 @@ namespace SolidStateGroup.BulletTrain
 {
     public class BulletTrainClient
     {
-        public string environmentKey { get; set; }
+        private readonly BulletTrainConfiguration configuration;
 
-        private static readonly HttpClient httpClient;
+        private static HttpClient httpClient;
+        private static bool isInitialized;
 
-        static BulletTrainClient() {
-            var sp = ServicePointManager.FindServicePoint(new Uri(Config.API));
-            sp.ConnectionLeaseTimeout = 60*1000*5;
-            httpClient = new HttpClient();
+        public BulletTrainClient(BulletTrainConfiguration bulletTrainConfiguration)
+        {
+            if (bulletTrainConfiguration == null)

```

Comment:
@kpdowns It'd be great here if after this null check you could check and throw errors for when the EnvironmentKey and ApiUrl is null or empty

Reply:
Hi @ssg-luke . Happy to make that change. What would you prefer? We could encapsulate the validation into the `BulletTrainConfiguration` object? I.e.

```
    public class BulletTrainConfiguration
    {
        public string ApiUrl = "https://api.bullet-train.io/api/v1/";
        public string EnvironmentKey = "";

        public void Validate() 
        {
            if(string.IsNullOrEmpty(ApiUrl))
            .....
        }
    }
```

Or simply doing it inline after that initial null check?

Going down the 1st path would keep the logic for validating the `BulletTrainConfiguration` coupled with an instance of the `BulletTrainConfiguration`. If the object then happens to become larger in future, it would be possible to add more checks in only one place, and it would technically allow you to validate the configuration before you try to create an instance of the client.

Reply:
1st path sounds good alongside calling that Validate() func after the initial null check in the BulletTrainClient constructor. I believe if Validate() then throws an exception it should also cause failure to construct the client without needing to necessarily ever call the Validate() func on the config object or alternatively have Validate() return bool and throw a generic exception for bad config in the constructor

Reply:
Ok great, I'm inclined to go for having `Validate` return a bool. Then after the null check when creating the client, call that. If it's not valid, then throw an exception during construction of the client. I'll make that change shortly.

Reply:
Updated, let me know if you'd like any other changes.

Reply:
Nope its looking good! Will get a v2.0 out when I find some time

Reply:
Thanks @ssg-luke !